### ISSUE TITLE
When open persistent with multiple db in one server, wrong db selected

### ISSUE THREAD

**Issue by Rivsen** [nrk/predis#577]
Example code:

```php
<?php

require './vendor/autoload.php';

$db0 = new Predis\Client([
    'host' => '127.0.0.1',
    'port' => 6379,
    'password' => null,
    'database' => 0,
    'persistent' => true,
], [
    'timeout' => 10,
]);

$db0->set('hello', 'world0');

$db1 = new Predis\Client([
    'host' => '127.0.0.1',
    'port' => 6379,
    'password' => null,
    'database' => 1,
    'persistent' => true,
], [
    'timeout' => 10,
]);

$db1->set('hello', 'world1');
$db0->set('hello', 'world00');
```

Redis monitor log shows this:

```
1567084084.622019 [0 172.29.0.7:31282] "SELECT" "0"
1567084084.647108 [0 172.29.0.7:31282] "SET" "hello" "world0"
1567084084.648049 [0 172.29.0.7:31282] "SELECT" "1"
1567084084.649324 [1 172.29.0.7:31282] "SET" "hello" "world1"
1567084084.650242 [1 172.29.0.7:31282] "SET" "hello" "world00"
```

The right behavior must be the string 'world00' write into db0 not db1.

**Comment by hhxsv5** [comment]
[phpredis](https://github.com/phpredis/phpredis) executes correctly with the different connection.
```php
<?php
$db0 = new Redis();
$db0->pconnect('127.0.0.1', 6379);
$db0->select(0);

$db0->set('hello', 'world0');

$db1 = new Redis();
$db1->pconnect('127.0.0.1', 6379);
$db1->select(1);

$db1->set('hello', 'world1');
$db0->set('hello', 'world00');
```
![image](https://user-images.githubusercontent.com/7278743/63951124-c97cd600-caaf-11e9-9d1c-edb5673f9f67.png)

So it is a bug?

**Comment by nrk** [comment]
Hi @Rivsen,

Unfortunately this is a limitation that can't be easily circumvented in userland code by Predis. Phpredis in contrast is a native C extension and that allows to do way more stuff under the hood to keep track of such things by relying on the Zend API.

That said, we provide a workaround for such cases by using a unique string instead of a simple boolean as the value for the `persistent` client option. You can find details on how this works with Predis v1.1 in commit 8277afc, additionaly #139 is the original issue that dates back to v0.8 although the client required a different configuration back then.

Basically the following snippet will open two different persistent connections when using Predis >= v1.1:

```php
$db0 = new Predis\Client([
    'database' => 0,
    'persistent' => 'db2',
]);

$db0->set('hello', 'world0');

$db1 = new Predis\Client([
    'database' => 1,
    'persistent' => 'db1',
]);

$db1->set('hello', 'world1');
$db0->set('hello', 'world00');
```
